File: VB/15_Document/DeleteLayer.vb
Imports Spire.Pdf

Namespace DeleteLayer
    Partial Public Class Form1
        Inherits Form
        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
            ' Create a new instance of the PdfDocument class.
            Dim doc As New PdfDocument()

            ' Load a PDF file from the specified path.
            doc.LoadFromFile("..\..\..\..\..\..\Data\DeleteLayer.pdf")

            ' Remove a layer named "red line" from the document's layers.
            doc.Layers.RemoveLayer("red line")

            ' Save the modified document to a file named "Output.pdf".
            doc.SaveToFile("Output.pdf")

            ' Close the document, releasing any resources associated with it.
            doc.Close()

            ' View the Pdf file
            PDFDocumentViewer("Output.pdf")
        End Sub
        Private Sub PDFDocumentViewer(ByVal fileName As String)
            Try
                Process.Start(fileName)
            Catch
            End Try
        End Sub
    End Class
End Namespace

File: CS/15_Document/DeleteLayer.cs
using Spire.Pdf;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace DeleteLayer
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Load the document from disk
            PdfDocument doc = new PdfDocument();
            doc.LoadFromFile(@"..\..\..\..\..\..\Data\DeleteLayer.pdf");

            // Remove the "red line" layer from the document
            doc.Layers.RemoveLayer("red line");

            // Save the modified document to a new file
            doc.SaveToFile("Output.pdf");

            // View the Pdf file
            PDFDocumentViewer("Output.pdf");
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
